AccessibilitySnapshot/Classes/FBSnapshotTestCase+Accessibility.swift

```diff
@@ -44,6 +44,19 @@ extension FBSnapshotTestCase {
         file: StaticString = #file,
         line: UInt = #line
     ) {
+        // The tests must be run in a host application in order for the accessibility properties to be populated
+        // correctly. The `UIApplication.shared` singleton is non-optional, but will be uninitialized when the tests are
+        // running outside of a host application, so we can use this check to determine whether we have a test host.
+        let hostApplication: UIApplication? = UIApplication.shared
+        guard hostApplication != nil else {
+            XCTFail(
+                "Accessibility snapshot tests cannot be run in a test target without a host application",
+                file: file,
+                line: line
```

Comment:
Something of a bit, but I’d consider originating this failure from your code, rather than from the calling file/line. This error isn’t caused by the call site but rather the configuration of the test.

Reply:
I could go either way on this, but I think it's still helpful to see which snapshot test failed. For example, if you want to temporarily disable the verify calls that are failing, it's helpful to highlight which ones are requiring the test host (rather than at a test level, since you could mix standard snapshot calls and accessibility snapshot calls in the same test). Given the descriptive failure message, I don't think it's that useful to know where in the framework the problem occurred, since ideally you don't have to know anything under the public API.

Going to merge as-is for now, but happy to keeping chatting and follow-up with a change if you disagree.

Reply:
Got it. I was imagining a world where accessibility snapshot tests were running as their own target. Makes sense that they might be mixed.